From cc65f02a6d3f8e3342c051f50432bb7b5c160763 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:51:58 -0400
Subject: [tools] perf/trace: Pack 'struct trace'

Message-id: <1408953210-26343-231-git-send-email-jolsa@redhat.com>
Patchwork-id: 88930
O-Subject: [PATCH RHEL7.1 BZ1133083 230/322] perf trace: Pack 'struct trace'
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 98eafce6bda705529e90d33b7f23706b346a6c8a
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Mon Jan 6 15:43:02 2014 -0300

description
===========
Initial struct stats:

 /* size: 368, cachelines: 6, members: 24 */
 /* sum members: 353, holes: 3, sum holes: 15 */
 /* last cacheline: 48 bytes */

After reorg:

[acme@ssdandy linux]$ pahole -C trace ~/bin/trace | tail -4
 /* size: 360, cachelines: 6, members: 24 */
 /* padding: 7 */
 /* last cacheline: 40 bytes */
};
[acme@ssdandy linux]$
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-trace.c b/tools/perf/builtin-trace.c
index 399b4b9..4bd44ab 100644
--- a/tools/perf/builtin-trace.c
+++ b/tools/perf/builtin-trace.c
@@ -1160,26 +1160,27 @@ struct trace {
  struct record_opts opts;
  struct machine  *host;
  u64   base_time;
- bool   full_time;
  FILE   *output;
  unsigned long  nr_events;
  struct strlist  *ev_qualifier;
- bool   not_ev_qualifier;
- bool   live;
  const char   *last_vfs_getname;
  struct intlist  *tid_list;
  struct intlist  *pid_list;
+ double   duration_filter;
+ double   runtime_ms;
+ struct {
+  u64  vfs_getname,
+    proc_getname;
+ } stats;
+ bool   not_ev_qualifier;
+ bool   live;
+ bool   full_time;
  bool   sched;
  bool   multiple_threads;
  bool   summary;
  bool   summary_only;
  bool   show_comm;
  bool   show_tool_stats;
- double   duration_filter;
- double   runtime_ms;
- struct {
-  u64  vfs_getname, proc_getname;
- } stats;
 };
 
 static int trace__set_fd_pathname(struct thread *thread, int fd, const char *pathname)
-- 
1.7.1